### PR TITLE
prepare v0.5.11 release

### PR DESCRIPTION
**Pull request by nnmin-aws**
This PR cherry picks PR merged on master branch to branch release-0.5.

It picked up the following pr in summary:
1. e2e test with kops
2. print version and commitId on startup
3. new dynamicfile backend mode
4. e2e test with kind

### CODE REVIEW

**Comment by nckturner** [comment]
Why is [1acb533](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/1acb533fc8ec789fb7c1d9ffdc7f22d588240742) included?

File: hack/start-dev-env-dynamicfile.sh [hunk 1]
```diff
@@ -56,6 +56,7 @@ write_kind_config
 create_kind_cluster
 certificate_authority_data="$(extract_certificate_authority_data)"
 write_kubectl_kubeconfig
+echo "start end to end testing for mountfile mode"
```
Comment (nckturner) [inline hack/start-dev-env-dynamicfile.sh:59]: My only nit is this is called "start dev env" but its used to run e2e tests.  Do you use this for both?  I'd like to separate them and have e2e tests run from the e2e script and dev envs set up from the dev script, and just have options to choose between dynamic file or other backends.  We can do that in a future improvement if you'd like.

**Comment by nnmin-aws** [comment]
> Why is [1acb533](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/1acb533fc8ec789fb7c1d9ffdc7f22d588240742) included?

thanks @nckturner for the comments. 
1acb533 is included because dynamicfile mode originally invoked two methods (validate) which belong to PR #416, this commit replaced the invocation of two methods.

File: hack/start-dev-env-dynamicfile.sh (hunk 1 above)
Comment (nnmin-aws) [reply hack/start-dev-env-dynamicfile.sh:59]: thanks @nckturner! this is a good point. I will create a separate PR after this cherry-pick to move the e2e out.

**Comment by nckturner** [comment]
For PR description, please make any changes if I've made a mistake:

[73a09f0](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/73a09f0b27bba45e5e34a3afa0293ea944d4d957), [c6a2c08](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/c6a2c08f1d4a57f60df8d0f658d7a6190db8ff96), [af5db65](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/af5db651192119baf72b9fe59c7c79c3f63c7819), [8684614](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/8684614bf818bdc571164efebcb251af2aeafabc), [bdfc6d2](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/bdfc6d26d0b6d483a1b97112e81f7bc526c0d56c): These commits are included to enable e2e testing on this branch.

[5d89d57](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/5d89d57c3d1307931891e2a5b1a2f83de1b54515): Minor bugfix related to dev env file permissions.

[45777aa](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/45777aa3e5d955cb48be2d88fb8e4f70b361eed8): This feature is being cherry picked to release-0.5 to support a feature launch in EKS--some platform versions use the release-0.5 branch and we need feature support in those versions.

[da8510b](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/da8510b9b9fbc638140e20b35f9733ae249a1423), [bcdcb3a](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/bcdcb3a4376a9acb891b6a79829fb4fe6b4a9c89): Print version on startup to make debugging easier.

[14e71c2](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/14e71c24d9ecf1e29cf8aa2c6c240be8656b6517): E2E test for 45777aa.

[ce87d8a](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/ce87d8ac7f25b273b1b6005f11d67b51778ce0f8): Update go version used to build.

[1acb533](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/1acb533fc8ec789fb7c1d9ffdc7f22d588240742): Remove dependency of 45777aa on the SSO PR.

[052daeb](https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/510/commits/052daeb25768400cd4a67adad16a337e28bcc493): Add makefile target for new e2e test.

**Pull request merged**